src/Microsoft.Azure.ServiceBus/MessageHandlerOptions.cs

```diff
@@ -75,6 +76,23 @@ public TimeSpan MaxAutoRenewDuration
             }
         }
 
+        /// <summary>Gets or sets the maximum number of messages the message pump should receive.</summary>
+        /// <value>The maximum number of messages.</value>
+        public int MaxMessageCount
```

Comment:
For a single message handler `MaxMessageCount` will be meaningless.
I'm concerned about the confusion it might introduce if the same `MessageHandlerOptions` is re-used for a single message handler and a batch handler.